etna/whatson/models.py

```diff
@@ -0,0 +1,633 @@
+        null=True,
+    )
+
+    speaker_image = models.ForeignKey(
+        get_image_model_string(),
+        null=True,
+        blank=False,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+
+    panels = [
+        FieldPanel("speaker_name"),
+        FieldPanel("speaker_description"),
+        FieldPanel("speaker_image"),
+    ]
+
+
+class EventSession(Orderable):
+    """
+    This model is used to add sessions to an event
+    e.g. 28th September @ 9:00, 29th September @ 10:30, 30th September @ 12:00.
+    These will link to the Eventbrite page.
+    """
+
+    page = ParentalKey(
+        "wagtailcore.Page",
+        on_delete=models.CASCADE,
+        related_name="event_sessions",
+    )
+
+    session_start_date = models.DateTimeField(
+        verbose_name=_("session start date"),
+        null=True,
+        blank=False,
+        help_text=_("The date and time the session starts."),
+    )
+
+    session_end_date = models.DateTimeField(
+        verbose_name=_("session end date"),
+        null=True,
+        blank=False,
+        help_text=_("The date and time the session ends."),
+    )
+
+    panels = [
+        FieldPanel("session_start_date"),
+        FieldPanel("session_end_date"),
+    ]
+
+
+class WhatsOnPage(BasePageWithIntro):
+    """WhatsOnPage
+
+    A page for listing events.
+    """
+
+    @cached_property
+    def events(self):
+        """
+        Returns a queryset of events that are children of this page.
+        """
+        return (
+            self.get_children()
+            .type(EventPage)
+            .specific()
+            .live()
+            .public()
+            .order_by("eventpage__start_date")
+        )
+
+    # DataLayerMixin overrides
+    gtm_content_group = "What's On"
+
+    class Meta:
+        verbose_name = _("What's On page")
+
+    parent_page_types = [
+        "home.HomePage",
+    ]
+    subpage_types = [
+        "whatson.EventPage",
+    ]
+
+    content_panels = BasePageWithIntro.content_panels
+
+
+class EventPage(ArticleTagMixin, TopicalPageMixin, BasePageWithIntro):
+    """EventPage
+
+    A page for an event.
+    """
+
+    # Content
+    lead_image = models.ForeignKey(
+        get_image_model_string(),
+        null=True,
+        blank=False,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+
+    # Event information
+    event_type = models.ForeignKey(
+        EventType,
+        null=True,
+        blank=False,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+
+    # Start and end date will be brought in from the API when we have it.
+    start_date = models.DateTimeField(
+        verbose_name=_("start date"),
+        null=True,
+        blank=True,
+        help_text=_("The date and time the event starts."),
+    )
+
+    end_date = models.DateTimeField(
+        verbose_name=_("end date"),
+        null=True,
+        blank=True,
+        help_text=_("The date and time the event ends."),
+    )
```

Comment:
If these fields are never for direct editing, it's usually clearer to add `editable=False`

```suggestion
    start_date = models.DateTimeField(
        verbose_name=_("start date"),
        null=True,
        editable=False,
    )

    end_date = models.DateTimeField(
        verbose_name=_("end date"),
        null=True,
        editable=False,
    )
```